### PR TITLE
Prevent the Analytics 4 module from being displayed unexpectedly in t…

### PR DESCRIPTION
**Pull request by sashadoes**
…he UI.

## Summary

Addresses issue:

- #6446 

## PR Author Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 5.2 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

File: assets/js/googlesitekit/modules/datastore/modules.js [hunk 1]
```diff
@@ -1343,6 +1343,28 @@ const baseSelectors = {
...
 
+	/**
+	 * Gets the list of internal shareable modules for dashboard sharing.
+	 *
+	 * Returns an Object/map of objects, keyed by slug as same as `getModules`.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state Data store's state.
+	 * @return {(Object|undefined)} Internal shareable modules available on the site; `undefined` if not loaded.
```
Comment (tofumatt) [inline assets/js/googlesitekit/modules/datastore/modules.js:1354]: The documentation for this function says this returns all _internal_ sharable modules, but in fact the function filters out modules that are internal and only returns the non-internal ones.

The behaviour here is correct, but the documentation is wrong—can you please update it? 🙂

File: assets/js/googlesitekit/modules/datastore/modules.test.js [hunk 2]
```diff
@@ -45,14 +45,54 @@ describe( 'core/modules modules', () => {
 		sharedOwnershipModules: [ 'analytics', 'search-console', 'tagmanager' ],
 	};
 
-	const recoverableModuleFixtures = [
+	const moduleFixturesAll = [
```
Comment (tofumatt) [inline assets/js/googlesitekit/modules/datastore/modules.test.js:48]: We can probably just drop the "all" suffix here; the `moduleFixturesExternal` suffix makes sense but I think here it's better to leave it out.

```suggestion
	const moduleFixtures = [
```

Or change the variables to reflect how you'd talk about them in English, eg:

```suggestion
	const allModulesFixture = [
```

File: assets/js/googlesitekit/modules/datastore/modules.test.js [hunk 3]
```diff
@@ -1983,5 +2025,65 @@ describe( 'core/modules modules', () => {
...
+				await untilResolved( registry, CORE_MODULES ).getModules();
+
+				const shareableModules = registry
+					.select( CORE_MODULES )
+					.getShareableModules();
+
+				expect( shareableModules ).toMatchObject( {} );
+			} );
+
+			it( 'should return the modules object for each internal shareable module', () => {
```
Comment (tofumatt) [inline assets/js/googlesitekit/modules/datastore/modules.test.js:2071]: Similar to the documentation note above, this is checking to ensure every module returned is _not_ internal (which is the correct behaviour), so this test spec should be updated.

**Review by tofumatt — CHANGES_REQUESTED** [review]
The code looks good, but the documentation/test specs here aren't accurate and need to be updated.

File: assets/js/googlesitekit/modules/datastore/modules.test.js (hunk 2 above)
Comment (sashadoes) [reply assets/js/googlesitekit/modules/datastore/modules.test.js:48]: Will go with `allModulesFixture ` cuz `modulesFixture` is already in use

**Comment by sashadoes** [comment]
Thx @tofumatt, I made an update 🙂

**Review by tofumatt — COMMENTED** [review]
There's a test here that isn't testing the right behaviour, but I've made the needed changes.

I've also changed `getSharedableModules` to return an object instead of an array, similar to `getModules`.

**Pull request merged**